pkg/backend/httpstate/client/client.go

```diff
@@ -1138,6 +1139,43 @@ func (pc *Client) GetCapabilities(ctx context.Context) (*apitype.CapabilitiesRes
 	return &resp, nil
 }
 
+func getSearchPath(orgName string) string {
+	return fmt.Sprintf("/api/orgs/%s/search/resources", orgName)
+}
+
+func getNaturalLanguageSearchPath(orgName string) string {
+	return fmt.Sprintf("/api/orgs/%s/search/resources/parse", orgName)
+}
+
+// Pulumi Cloud Search Functions
+func (pc *Client) GetSearchQueryResults(
+	ctx context.Context, orgName string, queryParams *apitype.PulumiQueryRequest,
+) (*apitype.ResourceSearchResponse, error) {
+	var resp apitype.ResourceSearchResponse
+	err := pc.restCall(ctx, http.MethodGet, getSearchPath(orgName), *queryParams, nil, &resp)
+	if err != nil {
+		return nil, fmt.Errorf("querying search failed: %w", err)
+	}
+	return &resp, nil
+}
+
+func (pc *Client) GetNaturalLanguageQueryResults(
+	ctx context.Context, orgName string, queryString string,
+) (*apitype.PulumiQueryResponse, error) {
+	var resp apitype.PulumiQueryResponse
+	var cleanedResponse apitype.PulumiQueryResponse
+	queryParamObject := apitype.PulumiQueryRequest{
+		Query: queryString,
+	}
+	err := pc.restCall(ctx, http.MethodGet, getNaturalLanguageSearchPath(orgName), queryParamObject, nil, &resp)
+	if err != nil {
+		return nil, fmt.Errorf("querying search failed: %w", err)
+	}
+	cleanedResponse.Query = strings.ReplaceAll(resp.Query, "\"", "")
```

Comment:
I'd like to better understand the motivation behind this step. What are we actually getting back from upstream?

Reply:
This endpoint parses a natural-language query like 

> all of my production vpcs

and returns search query syntax like

> type:"aws:ec2/vpc:Vpc" stack:production

(so the actual response payload is `{"query":"type:\"aws:ec2/vpc:Vpc\" stack:production"}`).

If I had to guess, this might be because `type:aws:ec2/vpc:Vpc` and `type:"aws:ec2/vpc:Vpc"` have different search semantics? (The latter is an exact match, the former allows for some partial matching.)

@kpitzen if that's the case, LMK why the non-quote semantics are preferred here. We can probably tweak the API slightly to make it behave better.

Edit: or this might simply be a misunderstanding -- that the quotes aren't meaningful when in fact they are.

Reply:
Initially, this was a workaround due to some inconsistencies between what I was sending the API directly via the standard query syntax and a similar query provided to the `parse` endpoint. I think that issue was actually caused and resolved elsewhere, and I never returned here to clean it up - another good catch :)